FWCore/Framework/interface/ProductRegistryHelper.h

```diff
@@ -43,6 +43,7 @@ namespace edm {
       TypeID typeID_;
       std::string productInstanceName_;
       std::string branchAlias_;
+      std::string switchAlias_;
```

Comment:
Do we really need two strings? Can they both ever be set? Maybe just an enum to say what type of alias? Could hid access to it behind a member function.